src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/Collections.generated.txt

```diff
@@ -89,9 +126,41 @@ namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
 
             foreach (IConfigurationSection section in configuration.GetChildren())
             {
-                if (section.Value is string stringValue4)
+                if (section.Value is string stringValue5)
+                {
+                    temp.Add(ParseInt(stringValue5, () => section.Path)!);
+                }
+            }
+        }
+
+        public static void BindCore(IConfiguration configuration, ref Dictionary<string, int> obj, BinderOptions? binderOptions)
+        {
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            foreach (IConfigurationSection section in configuration.GetChildren())
+            {
+                if (section.Value is string stringValue6)
+                {
+                    obj[section.Key!] = ParseInt(stringValue6, () => section.Path)!;
+                }
+            }
+        }
+
+        public static void BindCore(IConfiguration configuration, ref IDictionary<string, int> obj, BinderOptions? binderOptions)
```

Comment:
```suggestion
        public static void BindCore(IConfiguration configuration, IDictionary<string, int> obj, BinderOptions? binderOptions)
```

Why does this need to be `ref`, we are never setting the parameter in the method.

Reply:
I believe I needed this for structs. I'll create a follow up to remove it for ref types.